swift/apple/FirezoneKit/Sources/FirezoneKit/Views/GrantVPNView.swift

```
@@ -17,7 +17,13 @@ final class GrantVPNViewModel: ObservableObject {
 
   func grantPermissionButtonTapped() {
     Log.app.log("\(#function)")
-    store.createVPNProfile()
+    Task {
+      do {
+        try await store.createVPNProfile()
+      } catch {
+        Log.app.error("\(#function): \(error)")
```

We were swallowing these silently before. All of these logger calls will be improved with #7560 and #7004

---

What happens to un-caught exceptions in UI callbacks? Swift just swallows them silently??

---

In this particular case, nothing would have happened (not even logging).

For SwiftUI callbacks (like a button click) the closure signature is `@MainActor () -> Void` so the compiler will prevent us from throwing inside those.